src/cloudWatchLogs/commands/searchLogGroup.ts

```diff
@@ -14,26 +14,44 @@ import {
 } from '../registry/logStreamRegistry'
 import { DataQuickPickItem } from '../../shared/ui/pickerPrompter'
 import { Wizard } from '../../shared/wizards/wizard'
-import { createInputBox } from '../../shared/ui/inputPrompter'
+import { createInputBox, InputBoxPrompter } from '../../shared/ui/inputPrompter'
 import { createURIFromArgs } from '../cloudWatchLogsUtils'
 import { DefaultCloudWatchLogsClient } from '../../shared/clients/cloudWatchLogsClient'
 import { CloudWatchLogs } from 'aws-sdk'
 import { LogGroupNode } from '../explorer/logGroupNode'
 import { RegionSubmenu, RegionSubmenuResponse } from '../../shared/ui/common/regionSubmenu'
+import { integer } from 'aws-sdk/clients/cloudfront'
+import { Prompter, PromptResult } from '../../shared/ui/prompter'
+import { createQuickPick, QuickPickPrompter } from '../../shared/ui/pickerPrompter'
+import { ItemLoadTypes } from '../../shared/ui/pickerPrompter'
+import { isValidResponse } from '../../shared/wizards/wizard'
+import { StepEstimator } from '../../shared/wizards/wizard'
 import { highlightDocument } from '../document/logStreamDocumentProvider'
 import { CancellationError } from '../../shared/utilities/timeoutUtils'
 import { localize } from 'vscode-nls'
 import { getLogger } from '../../shared/logger'
 
 function handleWizardResponse(response: SearchLogGroupWizardResponse, registry: LogStreamRegistry): CloudWatchLogsData {
-    const logGroupInfo = {
+    const logGroupInfo: CloudWatchLogsGroupInfo = {
         groupName: response.submenuResponse.data,
         regionName: response.submenuResponse.region,
     }
+    let parameters: CloudWatchLogsParameters
+    const limitParam = registry.configuration.get('limit', 10000)
 
-    const parameters: CloudWatchLogsParameters = {
-        limit: registry.configuration.get('limit', 10000),
-        filterPattern: response.filterPattern,
+    if (response.timeRange.start === response.timeRange.end) {
+        // this means no time filter.
+        parameters = {
+            limit: limitParam,
+            filterPattern: response.filterPattern,
+        }
+    } else {
+        parameters = {
```

Comment:
could use a ternary, then `parameters` could be `const`. That also improves readability by avoiding nesting.